File: 5440-8312-443-string-compression/5440-8312-443-string-compression.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function(chars) {
    let index = 0 //,ans = []

    for(let i=0;i<chars.length;){
        let count = 0,ch = chars[i]

        while(i<chars.length && chars[i]===ch){
            count++;
            i++;
        }

        chars[index] = ch
        //ans[index] = ch
        index++;

        if(count>1){
            temp = count.toString().split('')
            for(let t of temp){
                chars[index] = t
                // ans[index] = t
                index++
            }
        }
    }
    // chars.length = 0
    // chars.push(...ans)
    return index;
};

File: 5440-8312-443-string-compression/5440-8312-443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        index,i = 0,0

        while i<len(chars):
            ch,count = chars[i],0

            while(i<len(chars) and chars[i]==ch):
                count+=1
                i+=1
        
            chars[index] = ch
            index+= 1

            if(count>1):
                for num in str(count):
                    chars[index] = num
                    index+= 1
                
        return index